tests/integration/Core/Framework/Plugin/PluginLifecycleServiceTest.php

```diff
@@ -490,10 +490,18 @@ public function testInstallationOfPluginWhichExecutesComposerCommandsWithPreviou
 
         $pluginWithExecuteComposer = $this->pluginService->getPluginByName('SwagTestExecuteComposerCommands', $this->context);
 
-        // Expected fail on executing the composer command, as the plugin is not in the default plugin directory and could therefore not be found
-        $this->expectException(PluginComposerRequireException::class);
-        $this->expectExceptionMessageMatches('/Your requirements could not be resolved to an installable set of packages/');
-        $this->pluginLifecycleService->installPlugin($pluginWithExecuteComposer, $this->context);
+        $exceptionWasThrown = false;
+
+        try {
+            // Expected fail on executing the composer command, as the plugin is not in the default plugin directory and could therefore not be found
+            $this->pluginLifecycleService->installPlugin($pluginWithExecuteComposer, $this->context);
+        } catch (PluginComposerRequireException $exception) {
+            $exceptionWasThrown = true;
+            static::assertStringContainsString('Your requirements could not be resolved to an installable set of packages', $exception->getMessage());
+        }
+
+        static::assertTrue($exceptionWasThrown);
+        \ComposerAutoloaderInit04843bd693b1d8a5a811fc7265ceb190::getLoader()->unregister();
```

Comment:
Kinda ugly i know, but the vendor folder is hard coded in the fixtures so it shouldn't change

Or is there a better way to get the autoloader for a specific folder? Require does not work, as it is already required during install and require_once won't give me the autoloader, but only return true as it is already required

Reply:
as this is our fixture, we can also rename it to something meaningful and then it should be fine :+1:

Reply:
Feels that this would be a usecase for `RunInSeparateProcess`. Unless such autoloader deregistration happens also in the code 🤔